tests/test_int_penaltymodel_ecosystem.py

```diff
@@ -1,38 +1,44 @@
 import unittest
+from unittest import mock
+import tempfile
 
 import networkx as nx
 
 import dimod
 
 import penaltymodel.core as pm
-import penaltymodel.cache as pmc
-import penaltymodel.maxgap as maxgap
-
 
 class TestInterfaceWithCache(unittest.TestCase):
     def test_retrieval(self):
-        # put some stuff in the database
+        tmp_db_file = tempfile.NamedTemporaryFile().name
```

Comment:
nice to do this as a context manager, e.g. `with tempfile.NamedTemporaryFile() as tf:`

Reply:
using TemporaryDirectory() instead as NamedTemporaryFile() is giving me errors (I think the file gets locked and then  I'm unable to write to it)

Reply:
Looks like using [`NamedTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) here will fail on Windows:
>  Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later)

Reply:
Yes, I switched to TemporaryDirectory (though Windows tests still passed when I was using NamedTemporaryFile()...)